### PR TITLE
🐞 fix: add exact option for array name in useWatch.

### PR DESCRIPTION
**Pull request by rabbitson87**
An unwanted update occurred when using useWatch as the array name. So I used the exact option, but it didn't work with the array name. The current exact option only works when it is not an array, so the condition has been changed to work on the array. Please review the corresponding update. Thank you.

### CODE REVIEW

**Comment by bluebill1049** [comment]
thanks, i will review it soon. 🙏

**Comment by rabbitson87** [comment]
> thanks, i will review it soon. 🙏

Thank you for accepting my opinion :).

**Review by bluebill1049 — DISMISSED** [review]
So this PR would fix the array payload with exact right?

File: src/logic/shouldSubscribeByName.ts [hunk 1]
```diff
@@ -6,7 +6,10 @@ export default <T extends string | string[] | undefined>(
   exact?: boolean,
 ) =>
   exact && signalName
-    ? name === signalName
+    ? name === signalName ||
+      convertToArrayPayload(name).some(
```
Comment (bluebill1049) [inline src/logic/shouldSubscribeByName.ts:10]: perhaps for better perf, we just check if payload is Array or not?

Comment (rabbitson87) [reply src/logic/shouldSubscribeByName.ts:10]: First, make sure it is an array and that the factors in it match exactly the event name

Comment (bluebill1049) [reply src/logic/shouldSubscribeByName.ts:10]: `Array.isArray(signalName) ? x : y`, so we split check string and array.

Comment (rabbitson87) [reply src/logic/shouldSubscribeByName.ts:10]: I don't think it's a problem whether it's a string or an array without having to distinguish. What do you think happens when there's a problem?

Comment (bluebill1049) [reply src/logic/shouldSubscribeByName.ts:10]: save some perf without have running `.some` each time.

Comment (rabbitson87) [reply src/logic/shouldSubscribeByName.ts:10]: All right, so can you see what it'll be like this?

Before
```
name === signalName ||
      convertToArrayPayload(name).some(
        (currentName) => currentName && exact && currentName === signalName,
      )
```

After
```
name === signalName ||
      (typeof name !== 'string' &&
        Array.isArray(name) &&
        name.some(
          (currentName) => currentName && exact && currentName === signalName,
        ))
```

Comment (bluebill1049) [reply src/logic/shouldSubscribeByName.ts:10]: is `Array.isArray(name)` enough?

Comment (rabbitson87) [reply src/logic/shouldSubscribeByName.ts:10]: I think it's enough.

Comment (bluebill1049) [reply src/logic/shouldSubscribeByName.ts:10]: Sounds good, let's do that then, we can merge this in after the fix.

Comment (rabbitson87) [reply src/logic/shouldSubscribeByName.ts:10]: I'll fix it right away and merge it! thank you :)

Comment (bluebill1049) [reply src/logic/shouldSubscribeByName.ts]: 👍

**Review by bluebill1049 — APPROVED** [review]
thank you 🙏

**Pull request merged**